{{cookiecutter.project_name}}/pyproject.toml

```diff
@@ -12,22 +12,18 @@ license = "{{cookiecutter.license}}"
 readme = "README.md"
 include = ["README.md", "src/{{cookiecutter.__project_slug}}/schema", "project"]
 
-requires-python = ">=3.9"
+requires-python = ">=3.9,<4.0"
 
 dynamic = ["version"]
 
 dependencies = [
-  "linkml-runtime >=1.8.0",
+  "linkml-runtime >=1.1.24",
 ]
 
 [tool.poetry]
 requires-poetry = ">=2.0"
 version = "0.0.0"
 
-[tool.poetry.dependencies]
-python = "^3.9"
```

Comment:
How does poetry determine now which version to use during `poetry install`?

Reply:
`poetry env info` tells what Poetry will use. Poetry will try first the version it was installed with but offers more control if needed (see [managing evironments](https://python-poetry.org/docs/managing-environments/)).